ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto

```diff
@@ -0,0 +1,140 @@
+    rpc DeleteUser (DeleteUserRequest) returns (google.protobuf.Empty);
+
+    // List the all existing users.
+    rpc ListUsers (ListUsersRequest) returns (ListUsersResponse);
+    
+    // Grant a right to a user. Idempotent if the user already has the right.
+    rpc CreateUserRight (CreateUserRightRequest) returns (Right);
+    
+    // Revoke a right from a user. Fails if the user does not have the right.
+    rpc DeleteUserRight (DeleteUserRightRequest) returns (google.protobuf.Empty);
+    
+    // List the set of all rights granted to a user.
+    rpc ListUserRights (ListUserRightsRequest) returns (ListUserRightsResponse);
+}
+
+
+// Users and rights
+///////////////////
+
+// Users are are used to manage the rights given to authenticated Ledger API clients.
+// They are stored and managed per participant node.
+//
+// TLS client certificates authenticate the user specified by the certificates common name.
+// JWT tokens authenticate the user specified by the JWT tokens 'subj' (subject) claim.
+message User {
+    // The user identifier, which must be a non-empty string of at most 64 
+    // characters matching the regexp [a-z0-9-_.], i.e. lower-case characters, digits,
+    // dashes or dots. Double-dots, double-dashes, and double-underscores are not allowed.
+    string id = 1;
+
+    // TODO: add Kubernetes-style labels to users to simplify writing
+    // integration code.
+}
+
+// A right granted to a user.
+message Right {
+    // The user is allowed to administrate the participant node.
+    message ParticipantAdmin {};
+
+    // The user can authorize commands in the name of the given party.
+    message CanActAs {
+        string party = 1;
+    }
+
+    // The user can read ledger data visible to the given party.
+    message CanReadAs {
+        string party = 1;
+    }
+
+    oneof kind {
+        ParticipantAdmin participant_admin = 1;
+        CanActAs can_act_as = 2;
+        CanReadAs can_read_as = 3;
+    }
+}
+
+
+// RPC requests and responses
+/////////////////////////////
+
+message CreateUserRequest {
+    // The user to create.
+    User user = 1;
+
+    // TODO: add for convenience:
+    // The rights to be granted to the user on creation.
+    // repeated Right rights = 2;
+}
+
+message DeleteUserRequest {
+    string user_id = 1;
+}
+
+message ListUsersRequest {
+    // TODO: add pagination, cf. https://cloud.google.com/apis/design/design_patterns#list_pagination
+}
+
+message ListUsersResponse {
+    repeated User users = 1;
+}
+
+// Add the right to the set of rights given to the user.
+// Idempotent if the user already has the right.
+message CreateUserRightRequest {
+    string user_id = 1;
+    Right right = 2;
+}
+
+// Remove the right from the set of rights granted to the user.
+// Fails if the user does not have the right.
+message DeleteUserRightRequest {
+    string user_id = 1;
+    Right right = 2;
+}
+
+message ListUserRightsRequest {
+    string user_id = 1;
```

Comment:
If we mandate the presence of a token here and read the user from the token we probably make it easier for client applications to actually treat the token opaquely. Otherwise we’ll repeat the pattern we have in the JSON API where the token is decoded in the client to figure out the parameters to send to the ledger which makes it increasingly hard to change the token format.

Reply:
Good point. Addressed in https://github.com/digital-asset/daml/pull/11818/commits/8880bd081d46649833257196a6a8018e9862ef9b